pkl-core/src/main/java/org/pkl/core/externalreader/ModuleResolver.java

```diff
@@ -13,24 +13,19 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.pkl.core.module;
+package org.pkl.core.externalreader;
 
 import java.io.IOException;
 import java.net.URI;
 import java.util.List;
 import org.pkl.core.SecurityManager;
 import org.pkl.core.SecurityManagerException;
+import org.pkl.core.messaging.MessageTransport;
+import org.pkl.core.module.PathElement;
 
-public interface ExternalModuleResolver {
-
-  interface Spec {
-    boolean hasHierarchicalUris();
-
-    boolean isGlobbable();
-
-    boolean isLocal();
-
-    String scheme();
+public interface ModuleResolver {
+  static ModuleResolver of(MessageTransport transport, long evaluatorId) {
+    return new ModuleResolverImpl(transport, evaluatorId);
```

Comment:
There aren't any layering violations here, are there?

Since we've now shipped this in 0.27, I think let's keep this as-is, and have `org.pkl.core.externalreader.ModuleReaderSpec` implement interface `Spec` here.

Reply:
`ModuleReaderSpec` is a simple record and callers of `ModuleKeys.externalResolver` may need to instantiate it.
I think the `ModuleResolver.Spec` interface has no value (at least not anymore) and just obfuscates the public API.

This PR already makes breaking API changes. If it's important to make this particular change less breaking, perhaps record `ModuleReaderSpec` could be renamed to `ModuleResolver.Spec`. However, I doubt this trade of breaking changes has much benefit.

I prefer `ModuleReaderSpec` over `ModuleResolver.Spec` because it makes more sense in the `ReaderProcess` interface:

```
interface ReaderProcess {
  // used to return ModuleResolver.Spec, which I found confusing
  ModuleReaderSpec getModuleReaderSpec(String scheme) throws IOException;

  // used to return ResourceResolver.Spec, which I found confusing
  ResourceReaderSpec getResourceReaderSpec(String scheme) throws IOException;
}